ShopApp/ShopApp/Controller/HomeViewController.swift

```diff
@@ -30,18 +32,52 @@ class HomeViewController: UIViewController {
         let navigationController = self.navigationController
         navigationController?.setViewControllers([loginViewController], animated: true)
     }
-    
+
+    func addUser() async {
+        let ref = Database.database().reference()
+
+        guard let name = self.currentUser?.displayName as? String else { return }
+        guard let email = self.currentUser?.email as? String else { return }
+
+        do {
+            // Get data from users table
+            let dataRead = await ref.child("users").observeSingleEventAndPreviousSiblingKey(of: .value)
+
+            let data = dataRead.0.value as? [String: Any]
+
+            // Check if data is null
+            guard let data = data else {
+                return
+            }
+
+            let emailVal = email.replacingOccurrences(of: ".", with: " ")
+
+            // For every user in users table
+            for dataKeys in data.keys where dataKeys.elementsEqual(emailVal) {
+                return
+            }
+
+            let refValue = Database.database().reference().child("users/\(emailVal)")
+            try await refValue.updateChildValues(["name": name])
+        } catch {
+            print(error)
+        }
+    }
+
     override func viewDidLoad() {
-        super.viewDidLoad()
+        async {
+            super.viewDidLoad()
+            await self.addUser()
```

Comment:
You can only add this line in the "async" block

Reply:
I tried, but it doesn't woks the way as I want it to